### PR TITLE
eslint fixes

### PR DESCRIPTION
**Pull request by B2o5T**

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -3,65 +3,59 @@ module.exports = {
...
     '@typescript-eslint/no-empty-interface': 'off',
-    'prefer-arrow-callback': 'error',
     '@typescript-eslint/no-empty-function': 'off',
     '@typescript-eslint/no-var-requires': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
     '@typescript-eslint/no-non-null-assertion': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/ban-ts-ignore': 'off',
     '@typescript-eslint/ban-types': 'off',
-    'no-lonely-if': 'error',
```
Comment (B2o5T) [inline .eslintrc.cjs:25]: they are already in config

File: .eslintrc.cjs [hunk 2]
```diff
@@ -3,65 +3,59 @@ module.exports = {
...
   overrides: [
     {
-      files: ['**/tests/**/*.ts', '**/graphql-codegen-testing/**/*.ts', '*.spec.ts'],
+      files: ['website/**'],
+      extends: '@theguild/eslint-config/react',
+    },
+    {
+      files: ['packages/**'],
+      // todo: enable in v3
+      rules: { 'unicorn/prefer-node-protocol': 'off' },
```
Comment (B2o5T) [inline .eslintrc.cjs:42]: to be sure it's non-breaking refactoring

File: packages/graphql-codegen-cli/src/codegen.ts [hunk 3]
```diff
@@ -217,10 +217,8 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
                     : outputConfig.preset
                   : null;
 
-                if (preset) {
-                  if (preset.prepareDocuments) {
-                    outputSpecificDocuments = await preset.prepareDocuments(filename, outputSpecificDocuments);
-                  }
+                if (preset && preset.prepareDocuments) {
```
Comment (B2o5T) [inline packages/graphql-codegen-cli/src/codegen.ts:220]: this rule https://tortilla-hq.slack.com/archives/C03B470CJCT/p1673028464154629 will catch this case :)

File: packages/graphql-codegen-cli/src/codegen.ts [hunk 4]
```diff
@@ -352,10 +350,9 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
 
                           const process = async (outputArgs: Types.GenerateOptions) => {
                             const output = await codegen({
-                              ...{
-                                ...outputArgs,
-                                emitLegacyCommonJSImports: shouldEmitLegacyCommonJSImports(config, outputArgs.filename),
-                              },
+                              ...outputArgs,
+                              // @ts-expect-error todo: fix 'emitLegacyCommonJSImports' does not exist in type 'GenerateOptions'
+                              emitLegacyCommonJSImports: shouldEmitLegacyCommonJSImports(config, outputArgs.filename),
```
[the remark this answers is not in the document]
Comment (saihaj) [reply packages/graphql-codegen-cli/src/codegen.ts:355]: yeah I think we can add. Not sure why we didn't do it in past 🤔 @n1ru4l might know?

**Review by saihaj — APPROVED** [review]

**Pull request merged**